js/test/unit/ipc/serialization-tests.ts

```diff
@@ -37,6 +37,30 @@ function createTable<T extends TypeMap = any>(schema: Schema<T>, chunkLengths: n
     return generate.table(chunkLengths, schema).table;
 }
 
+describe('tableFromIPC', () => {
+    test('handles AsyncRecordBatchReader input', async () => {
+        type T = { a: Uint8 };
+
+        const sources = [
+            new Table({ a: makeVector(new Uint8Array([1, 2, 3])) }),
+            new Table({ a: makeVector(new Uint8Array([4, 5, 6])) }),
+        ];
+
+        const writer = sources.reduce(
+            (writer, source) => writer.writeAll(source),
+            new RecordBatchStreamWriter<T>({ autoDestroy: false })
+        );
+
+        writer.close();
+
+        let index = 0;
+
+        for await (const reader of RecordBatchReader.readAll<T>(writer[Symbol.asyncIterator]())) {
+            expect(sources[index++]).toEqualTable(await tableFromIPC(reader));
```

Comment:
Shouldn't this be the other way around (actual on the left)?

Reply:
Done